File: Baekjoon/해시/1620.py
# https://www.acmicpc.net/problem/1620

import sys
input = sys.stdin.readline

n, m = map(int, input().split())
pokemon = {}
pokemon_num = {}

for i in range(1, n + 1):
    name = input().strip()
    pokemon[i] = name
    pokemon_num[name] = i
    
for _ in range(m):
    q = input().strip()
    if q.isdigit():
        print(pokemon[int(q)])
    else:
        print(pokemon_num[q])

File: Baekjoon/해시/1620.cpp
// https://www.acmicpc.net/problem/1620

#include <iostream>
#include <map>
#include <string>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int N, M;
    cin >> N >> M;

    map<string, int> nameToIndex;
    map<int, string> indexToName;
    for (int i = 1; i <= N; i++) {
        string name;
        cin >> name;
        nameToIndex[name] = i;
        indexToName[i] = name;
    }

    for (int i = 0; i < M; i++) {
        string query;
        cin >> query;
        if (query[0] >= '0' && query[0] <= '9') {
            cout << indexToName[stoi(query)] << '\n';
        } else {
            cout << nameToIndex[query] << '\n';
        }
    }

    return 0;
}